hack/generated/controllers/controller_test.go

```diff
@@ -96,17 +97,77 @@ func Test_StorageAccount_CRUD(t *testing.T) {
 	g.Expect(acct.Status.Id).ToNot(BeNil())
 	armId := *acct.Status.Id
 
+	// Run sub-tests on storage account
+	t.Run("Blob Services CRUD", func(t *testing.T) {
+		StorageAccount_BlobServices_CRUD(t, testContext, acct.ObjectMeta)
+	})
+
 	// Delete the storage account
 	err = testContext.KubeClient.Delete(ctx, acct)
 	g.Expect(err).ToNot(HaveOccurred())
 	g.Eventually(acct).Should(testContext.Match.BeDeleted(ctx))
 
 	// Ensure that the resource group was really deleted in Azure
-	// TODO: Do we want to just use an SDK here? This process is quite icky as is...
 	exists, err := testContext.AzureClient.HeadResource(
 		ctx,
 		armId,
 		"2019-04-01")
 	g.Expect(err).ToNot(HaveOccurred())
 	g.Expect(exists).To(BeFalse())
 }
+
+func StorageAccount_BlobServices_CRUD(t *testing.T, testContext testcommon.KubePerTestContext, storageAccount metav1.ObjectMeta) {
+	ctx := context.Background()
+
+	g := NewGomegaWithT(t)
+
+	blobService := &storage.StorageAccountsBlobService{
+		ObjectMeta: ctrl.ObjectMeta{
+			Name:      "default", // MUST be 'default'
+			Namespace: testContext.Namespace(),
+		},
+		Spec: storage.StorageAccountsBlobServices_Spec{
+			ApiVersion: "2019-04-01", // TODO [apiversion]: to be removed eventually
+			Owner:      testcommon.AsOwner(storageAccount),
+		},
+	}
+
+	// Create
+	err := testContext.KubeClient.Create(ctx, blobService)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Eventually(blobService).Should(testContext.Match.BeProvisioned(ctx))
+
+	t.Run("Container CRUD", func(t *testing.T) {
+		StorageAccount_BlobServices_Container_CRUD(t, testContext, blobService.ObjectMeta)
+	})
+
+	// TODO: Delete doesn't seem to work?
```

Comment:
What's the alternative? Does deletion at the ResourceGroup level work? If it does might be worth saying that since we're not going to leak any resources at least...

Reply:
@matthchr This is owned by a StorageAccount resource, so the lifetime would be tied to that.

Reply:
Expanded comment